src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -526,7 +526,7 @@ void SystemDictionaryShared::handle_class_unloading(InstanceKlass* klass) {
     // Remove the class from _unregistered_classes_table: keep the entry but
```

Comment:
Pre-existing, but `_unregistered_classes_table` is set under the lock above, but here it is checked outside of the lock. Maybe MutexLocker should be move above the `if` and placed in a `{ ... }` so that it is unlocked before the `if (ClassListWriter:...)` statement?